qiskit/circuit/quantumcircuit.py

```diff
@@ -2565,8 +2567,27 @@ def add_calibration(self, gate, qubits, schedule, params=None):
         """
         if isinstance(gate, Gate):
             self._calibrations[gate.name][(tuple(qubits), tuple(gate.params))] = schedule
+            if params is not None:
+                self.add_calibration_params(gate.name, qubits, params)
         else:
             self._calibrations[gate][(tuple(qubits), tuple(params or []))] = schedule
+            if params is not None:
+                self.add_calibration_params(gate, qubits, params)
+
+    def add_calibration_params(self, name, qubits, params):
+        """Update parameters table with calibration parameters
+
+        Args:
+            name (str): Instruction name.
+            qubits (Union[int, Tuple[int]]): List of qubits to be measured.
+            params (Optional[List[Union[float, Parameter]]]): A list of parameters.
+        """
+        for instruction, qargs, _ in self._data:
+            for qarg in qargs:
+                for qbit in qubits:
+                    if instruction.name == name and qarg.index == qbit:
+                        instruction.params = params
```

Comment:
not sure about the side effects of this -- we might want to see if we can do this another way
@kdk ?

Reply:
What about the case where the instruction also has parameters, which are distinct from the schedule's parameters? Would this overwrite them?

Reply:
I'm not confident whether it's desirable for `self.assertTrue(alpha in circ.parameters)` (from tests) to be True when `alpha` appears in the calibrations